services/templeton/src/templeton/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,86 @@
+package elasticsearch
+
+import (
+	"logcabin"
+
+	"encoding/json"
+	"github.com/mattbaird/elastigo/lib"
+
+	"templeton/database"
+	"templeton/model"
+)
+
+var (
+	logger = logcabin.New()
+)
+
+// Elasticer is a type used to interact with Elasticsearch
+type Elasticer struct {
+	es      *elastigo.Conn
+	baseURL string
+	index   string
+}
+
+// NewElasticer returns a pointer to an Elasticer instance that has already tested its connection
+// by making a WaitForStatus call to the configured Elasticsearch cluster
+func NewElasticer(elasticsearchBase string, elasticsearchIndex string) (*Elasticer, error) {
+	c := elastigo.NewConn()
+
+	err := c.SetFromUrl(elasticsearchBase)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
As far as I can see, this is the idiomatic way to do it -- this exact sequence is literally just passing the same error up the chain. Go also lets you raise a panic (which is e.g. what the language itself does for indexes out of bounds) but that'd just change the piece inside the if, since I can't control how the libraries pass errors. I'm wondering if there's something nicer too, though, or at least more of an idea if I really need to be catching them all, what to do with them, etc. more than just passing them up the whole way and then logging them and dying. This particular one I think will just always be fatal, since it's not going to be able to do much with an elasticsearch base URL it can't parse, but I'm not sure that's generally true.